stripe-core/src/main/java/com/stripe/android/core/networking/SendAnalyticsRequestV2Worker.kt

```diff
@@ -0,0 +1,80 @@
+package com.stripe.android.core.networking
+
+import android.content.Context
+import androidx.annotation.VisibleForTesting
+import androidx.work.CoroutineWorker
+import androidx.work.Data
+import androidx.work.WorkerParameters
+import androidx.work.workDataOf
+import com.stripe.android.core.exception.APIException
+import com.stripe.android.core.exception.InvalidRequestException
+import kotlinx.serialization.encodeToString
+import kotlinx.serialization.json.Json
+
+private const val DataKey = "data"
+
+internal class SendAnalyticsRequestV2Worker(
+    appContext: Context,
+    params: WorkerParameters,
+) : CoroutineWorker(appContext, params) {
+
+    override suspend fun doWork(): Result = withRequest { request ->
+        val stripeNetworkClient = getNetworkClient()
+
+        return runCatching {
+            stripeNetworkClient.executeRequest(request)
+        }.fold(
+            onSuccess = {
+                Result.success()
+            },
+            onFailure = { error ->
+                if (error.shouldRetry) {
+                    Result.retry()
+                } else {
+                    Result.failure()
+                }
+            },
+        )
+    }
+
+    private inline fun withRequest(block: (AnalyticsRequestV2) -> Result): Result {
+        val request = getRequest(inputData) ?: return Result.failure()
+        return block(request)
+    }
+
+    companion object {
+
+        const val TAG = "SendAnalyticsRequestV2Worker"
+
+        private var networkClient: StripeNetworkClient? = null
+
+        fun getNetworkClient(): StripeNetworkClient {
```

Comment:
why `networkClient` is initialized this way? can you just set it as a constrctor parameter of `SendAnalyticsRequestV2Worker` with default value? that way you don't need a dedicated `setNetworkClient` for test